ddtrace/internal/telemetry/metrics.py

```diff
@@ -37,43 +36,32 @@ def __init__(self, namespace, name, tags, common, interval=None):
         common: set to True if a metric is common to all tracers, false if it is python specific
         interval: field set for gauge and rate metrics, any field set is ignored for count metrics (in secs)
         """
-        self.name = name
+        self.name = name.lower()
         self.is_common_to_all_tracers = common
         self.interval = interval
-        self.namespace = namespace
-        self._tags = {}  # type: MetricTagType
-        self.set_tags(tags)
+        self.namespace = namespace.lower()
+        self._tags = {k.lower(): str(v).lower() for k, v in tags.items()}
         self._count = 0.0
-        self._points = []  # type: List[float]
+        self._points = []  # type: List
 
-    @property
-    def id(self):
+    @classmethod
+    def get_id(cls, name, namespace, tags, metric_type):
+        # type: (str, str, Dict[str, Any], str) -> str
         """
         https://www.datadoghq.com/blog/the-power-of-tagged-metrics/#whats-a-metric-tag
         """
-        return self.name + str(self._tags)
+        str_tags = str(sorted(tags.items())) if tags else ""
```

Comment:
this isn't doing the same thing as `__init__` for `self._tags`, is that going to end up being a problem?

Reply:
e.g. `Metric.get_id(...)` won't be the same as `hash(metric)`

not sure I am asking for a change, just calling out the inconsistency, you are better equipped to decide if it is a problem or not.

Reply:
Thanks for the call out. This a problem. I think we should just avoid lowercasing tags and make sure the caller sets the correct values. I will update this and tests